gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -608,19 +611,20 @@ private final PreparedStatement prepMinion(
 						if (prepareIsolationLevel == Connection.TRANSACTION_NONE) {
 							cc.markAsDDLForSnappyUse(true);
 							return getPreparedStatementForSnappy(false, statementContext, lcc,
-                  cc.isMarkedAsDDLForSnappyUse(), checkCancellation);
+                  cc.isMarkedAsDDLForSnappyUse(), checkCancellation, false);
 						}
 					}
 					qt = p.parseStatement(getQueryStringForParse(lcc), paramDefaults);
 				}
 				catch (StandardException | AssertFailure ex) {
           //wait till the query hint is examined before throwing exceptions or
-          if (routeQuery && !DML_TABLE_PATTERN.matcher(source).matches()) {
+          if (routeQuery && !INSERT_TABLE_PATTERN.matcher(source).matches()) {
```

Comment:
Only at last routing is being done on basis of column table (added my comment below). At rest of the places this is just to handle any exception case where if update/delete get any exception on XD side then just route it to Snappy. So they can be both row and column table. We had a small discussion about same the other day.